### PR TITLE
ID Type Detection / Single ID Start

### LINKED ISSUE #9: Automatically identify IDTypes on import
IDTypes should be identified when importing a dataset automatically.

Genes: Ensembl ID (beginning with ENSG)
Cell Lines & Tissues: query database

First the columns are searched for the string "ENSG" to find Ensembl IDs. If a column with these substrings is found, it is considered as the ID column.

Otherwise the database is queried to find the IDs (e.g. for cell lines & tissues). If the IDs are found, this column is considered as the ID column again.

### PR DESCRIPTION
**Pull request by chrilehner**
closes #9

### CODE REVIEW

File: src/GeneIDTypeDetector.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+function detectIDType(data: any[], accessor: (row: any) => string, sampleSize: number) : Promise<number> {
+  return new Promise((resolve) => {
```
Comment (sgratzl) [inline src/GeneIDTypeDetector.ts]: why wrapping in a new Promise construct, when Promise.resolve would also do the job?

File: src/GeneIDTypeDetector.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        ++foundIDTypes;
+      }
+      ++validSize;
+    }
+
+    resolve(foundIDTypes / validSize);
+  });
+}
+
+export function geneIDTypeDetector() {
```
Comment (sgratzl) [inline src/GeneIDTypeDetector.ts]: is there an interface that descdribes what the output has to look like?

File: phovea.js [hunk 3]
```diff
@@ -5,70 +5,100 @@
  **************************************************************************** */
 
 //register all extensions in the registry following the given pattern
-module.exports = function(registry) {
+module.exports = function (registry) {
```
Comment (sgratzl) [inline phovea.js]: please try to format only your code and not the one of others

**Comment by sgratzl** [comment]
+ in general please try to avoid mixing two separate issues in one branch, in this case both the search of single gene and detecting id types

File: phovea.js (hunk 3 above)
Comment (chrilehner) [inline phovea.js]: sorry, I saw the same format in another phovea.js file and thought this one would be more readable. I won't reformat all the code in the future.

Comment (chrilehner) [inline phovea.js]: should I undo it?

**Pull request merged**